### PR TITLE
8265746: Update java.time to use instanceof pattern variable (part II)

### PR DESCRIPTION
**Pull request by pconcannon**
Hi,

Could someone please review the second half of my update for the `java.time` package to make use of the `instanceof` pattern variable?

This PR was split into two parts due to the large number of files affected.

Kind regards,
 Patrick

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265746](https://bugs.openjdk.java.net/browse/JDK-8265746): Update java.time to use instanceof pattern variable (part II)

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to 974ccbf07e223c857dee65da67abe22c0c367ffc
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3650/head:pull/3650` \
`$ git checkout pull/3650`

Update a local copy of the PR: \
`$ git checkout pull/3650` \
`$ git pull https://git.openjdk.java.net/jdk pull/3650/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3650`

View PR using the GUI difftool: \
`$ git pr show -t 3650`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3650.diff">https://git.openjdk.java.net/jdk/pull/3650.diff</a>

</details>

### CODE REVIEW

**Review by dfuch — APPROVED** [review]
LGTM!

**Review by LanceAndersen — APPROVED** [review]
Looks good.  Thank you for the updates

**Review by RogerRiggs — APPROVED** [review]

File: src/java.base/share/classes/java/time/Clock.java [hunk 1]
```diff
@@ -618,11 +618,9 @@ public Instant instant() {
...
         @Override
         public boolean equals(Object obj) {
-            if (obj instanceof FixedClock) {
-                FixedClock other = (FixedClock) obj;
-                return instant.equals(other.instant) && zone.equals(other.zone);
-            }
-            return false;
+            return (obj instanceof FixedClock other
+                    && instant.equals(other.instant)
+                    && zone.equals(other.zone));
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/time/Clock.java:623]: The outer set of braces is redundant.

File: src/java.base/share/classes/java/time/Clock.java [hunk 2]
```diff
@@ -670,11 +668,9 @@ public Instant instant() {
...
         @Override
         public boolean equals(Object obj) {
-            if (obj instanceof OffsetClock) {
-                OffsetClock other = (OffsetClock) obj;
-                return baseClock.equals(other.baseClock) && offset.equals(other.offset);
-            }
-            return false;
+            return (obj instanceof OffsetClock other
+                    && baseClock.equals(other.baseClock)
+                    && offset.equals(other.offset));
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/time/Clock.java:673]: The outer set of braces is redundant.

File: src/java.base/share/classes/java/time/ZonedDateTime.java [hunk 3]
```diff
@@ -2190,13 +2185,10 @@ public boolean equals(Object obj) {
...
-            ZonedDateTime other = (ZonedDateTime) obj;
-            return dateTime.equals(other.dateTime) &&
-                offset.equals(other.offset) &&
-                zone.equals(other.zone);
-        }
-        return false;
+        return (obj instanceof ZonedDateTime other
+            && dateTime.equals(other.dateTime)
+            && offset.equals(other.offset)
+            && zone.equals(other.zone));
```
Comment (ChrisHegarty) [inline src/java.base/share/classes/java/time/ZonedDateTime.java:2191]: same here.

**Review by ChrisHegarty — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

File: src/java.base/share/classes/java/time/Clock.java (hunk 1 above)
Comment (pconcannon) [reply src/java.base/share/classes/java/time/Clock.java]: Braces removed in 7f32ddb

File: src/java.base/share/classes/java/time/ZonedDateTime.java (hunk 3 above)
Comment (pconcannon) [reply src/java.base/share/classes/java/time/ZonedDateTime.java]: See 7f32ddb

**Pull request closed without merging**